kamailio/db_queries_postgres.cfg

```diff
@@ -6,9 +6,11 @@
 #!substdef "!KZQ_EVENT_PRESENCE_RESET_DELETE!delete from presentity where domain='\$(kzE{kz.json,Realm})' and username = '\$(kzE{kz.json,Username})'!g"
 #!substdef "!KZQ_HANDLE_NEW_SUBSCRIBE_DELETE1!delete from active_watchers where callid = '\$ci'!g"
 #!substdef "!KZQ_HANDLE_NEW_SUBSCRIBE_DELETE2!delete from active_watchers where watcher_username='\$fU' and presentity_uri='\$var(presentity_uri)' and to_user='\$tU' and watcher_domain='\$fd' and event='\$hdr(Event)'!g"
+#!substdef "!KZQ_HAS_PRESENTITY!select count(*) as count from presentity where username = '\$subs(to_user)' and domain = '\$subs(to_domain)' and event = '\$subs(event)'!g"
 #!substdef "!KZQ_RESET_ACCOUNT_DELETE!delete from presentity where domain='\$(kzE{kz.json,Realm})'!g"
 #!substdef "!KZQ_RESET_ACCOUNT_UPDATE!update active_watchers set expires = \$TS where watcher_domain='\$(kzE{kz.json,Realm})'!g"
 #!substdef "!KZQ_RESET_PUBLISHER_UPDATE!update active_watchers set expires = \$TS where id in (select b.id from presentity a inner join active_watchers b on a.username = b.to_user and a.domain = b.to_domain and a.event = b.event where a.sender = '\$var(MediaUrl)')!g"
 #!substdef "!KZQ_PRESENCE_SEARCH_DETAIL!select * from active_watchers_log where presentity_uri = '\$var(presentity_uri)'!g"
 #!substdef "!KZQ_PRESENCE_SEARCH_SUMMARY!select * from active_watchers where watcher_domain = '\$var(Domain)'!g"
 #!substdef "!KZQ_PRESENCE_RESET!delete from presentity where sender = '\$var(MediaUrl)'!g"
+#!substdef "!KZQ_REPLACE_WATCHERS_LOG! DELETE FROM active_watchers_log WHERE presentity_uri='\$subs(uri)' and watcher_username='\$subs(watcher_username)' and watcher_domain='\$subs(watcher_domain)' and event='\$subs(event)'; INSERT INTO active_watchers_log (presentity_uri, watcher_username, watcher_domain, event, callid, to_user, to_domain, user_agent, time, result, sent_msg, received_msg) VALUES ('\$subs(uri)', '\$subs(watcher_username)', '\$subs(watcher_domain)', '\$subs(event)', '\$subs(callid)', '\$subs(to_user)', '\$subs(to_domain)', '\$(subs(user_agent){s.escape.common}{s.replace,',''}{s.replace,\$\$,})', \$TS, \$notify_reply(\$rs), '\$(mb{s.escape.common}{s.replace,',''}{s.replace,\$\$,}{s.replace,\\\\\",\"})', '\$(notify_reply(\$mb){s.escape.common}{s.replace,',''}{s.replace,\$\$,})')!g"
```

Comment:
I added `{s.replace,\\\\\",\"}` that is removes `"` escaping in blob.
Is correct case when blob contains raw (not escaped) XML?

Reply:
Also here in macro defined two SQL commands `DELETE` and `INSERT`.
This may be replaced by one `INSERT` with [`ON CONFLICT Clause`](https://www.postgresql.org/docs/9.5/static/sql-insert.html#SQL-ON-CONFLICT).
In this case must be used PostgreSQL 9.5 or newer.